discussion/evaluation-of-project.tex

```diff
@@ -1,6 +1,80 @@
+  As an example, the Sensor Tracker will output samples over the \gls{usart} interface, these samples are converted to strings using the {\core} library \code{format!} macro.
+  Using this macro for the types \code{u8, i32, u32} increases the size of the optimized binary with 1736B.
+
+\item [There is zero size overhead of using {\rust}] \hfill \\
+  The Minimal Main application shows that using {\rust} by itself does not incur an increase in size.
+  It is the size of the software libraries that in our results increases the size of the {\rust} binaries
+
+\end{description}
+
+\subsection{Energy}
+
+We must emphasis that energy consumption measurements presented in \autoref{sec:res:energy} are preliminary and that the focus for the Sensor Tracker application was primarily to provide an application in this domain.
+The main problem with reading to much into these results, is that the {\rust} and {\C} implementations share a substantial amount of the library code, written in {\C}.
+The application code here is just a thin wrapper around the libraries.
+When we dug deeper into the analysis of the application we found that most of the time, the application spends its time in a busy-wait state, waiting for sensor data to be available.
+In these circumstances the difference between the two languages are not be evident.
+
+What we can take away from the results presented is that the variance in the energy consumption in the applications tested are larger within the different optimizations levels of the {\C} compiler, compared to introducing the {\rust} application layer.
+As we saw in \autoref{sec:res:energy:cache} the variance is more correlated to the cache hit ratio than what programming language we are using.
+This result at least points in the direction of the two language being equal in this domain.
+
+In order to evaluate the {\rust} language thoroughly in this domain, a larger portion of the code constituting the library must be written in {\rust}.
+In addition a more suitable example must be employed, where the actual code is the dominant factor and not the busy-waits.
+
+\subsection{Rust in Embedded Systems}
+
+The results discussed here are positive for further exploration of {\rust} in the embedded domain.
+One critical task in an embedded system is to meet a strict deadline imposed by timely interrupts.
+Both the Sensor Tracker and Circle Game expose similar performance for both language implementations and this provides an important result.
+
+On the other hand when we look at code size we see a larger challenge for the {\rust} language.
+The focus on \concept{Zero Cost Abstractions} seems not to take into account the size of the end binary.
+Especially not in the builds where optimizations are not applied.
+In a system where excess resources for storing code and data this does not cause any problems, but this is not the case in an embedded system.
```

Comment:
In a system ~~where~~ `with` excess resources for storing code and data`, `this does not cause any problems, but this is not the case in an embedded system.